local_server/local-server-generate-elevation-profile/src/main/java/com/esri/samples/local_server_generate_elevation_profile/LocalServerGenerateElevationProfileController.java

```diff
@@ -0,0 +1,569 @@
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Camera;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.LayerSceneProperties;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import com.esri.arcgisruntime.raster.HillshadeRenderer;
+import com.esri.arcgisruntime.raster.Raster;
+import com.esri.arcgisruntime.raster.RasterFunction;
+import com.esri.arcgisruntime.raster.RasterFunctionArguments;
+import com.esri.arcgisruntime.symbology.ColorUtil;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingFeatures;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingJob;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingString;
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingTask;
+
+public class LocalServerGenerateElevationProfileController {
+
+  @FXML private Button drawPolylineButton;
+  @FXML private Button generateProfileButton;
+  @FXML private Button clearResultsButton;
+  @FXML private Label instructionsLabel;
+  @FXML private ProgressBar progressBar;
+  @FXML private SceneView sceneView;
+  @FXML private VBox vBox;
+
+  private ArcGISScene scene;
+  private FeatureCollection featureCollection;
+  private GeoprocessingTask gpTask;
+  private GraphicsOverlay graphicsOverlay;
+  private LocalGeoprocessingService localGPService;
+  private Polyline polyline;
+  private Raster arranRaster;
+  private RasterLayer rasterLayer;
+  private Viewpoint rasterExtentViewPoint;
+
+  private static LocalServer server;
+
+  /**
+   * Called after FXML loads. Sets up scene.
+   */
+  public void initialize() {
+
+    try {
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a scene with a topographic basemap style
+      scene = new ArcGISScene(BasemapStyle.ARCGIS_HILLSHADE_DARK);
+
+      // create a new raster from local file and display it on the scene
+      arranRaster = new Raster(new File
+        (System.getProperty("data.dir"), "./samples-data/local_server/Arran_10m_raster.tif").getAbsolutePath());
+      displayRaster(arranRaster);
+
+      // set up a new feature collection
+      featureCollection = new FeatureCollection();
+
+      // create a graphics overlay for displaying the sketched polyline and add it to the scene view's list of 
+      // graphics overlays
+      graphicsOverlay = new GraphicsOverlay();
+      SimpleLineSymbol lineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DASH,
+        ColorUtil.colorToArgb(Color.BLACK), 3);
```

Comment:
I wonder if it could be more colourful?  
![image](https://user-images.githubusercontent.com/5185225/174581256-c5622f65-8e71-4ca8-9343-6b29d685f64a.png)

Reply:
Mike had the same comment! I felt the white stood out well against the grayscale backdrop: I can experiment with pale pastel colours but also intentionally want the 2D line of section to be muted as only a reference point and have the eye drawn to the elevation profile.